src/kqueue.rs

```diff
@@ -156,12 +156,41 @@ impl EventLoop {
                         */
                         kqueue::Vnode::Delete => {
                             remove_watches.push(path.clone());
-                            Event::new(EventKind::Remove(RemoveKind::Any))
+                            Event::new(EventKind::Remove(RemoveKind::Any)).add_path(path)
                         }
 
-                        //data was written to this file
+                        // a write to a directory mean that a new file was created in it, let's
+                        // figure out which file this was
+                        kqueue::Vnode::Write if path.is_dir() => {
+                            // find which file is new in the directory be comparing it the our
+                            // list of known watches
+                            if let Some(file) = std::fs::read_dir(&path)
+                                .unwrap()
```

Comment:
please don't unwrap here, see #368

Reply:
I've update the code at this point so that we don't cause a panic, however i'm unware of any way to deal with this error. For now I just ignore it and return a Modify event for the directory. 

Question on the side: could we maybe add the log crate?

Reply:
we can emit an error event like we do at other places via the channel

I can see the appeal of using logging, but it's also pretty invasive as we don't use this anywhere else and do have a back channel for emitting errors. Though it probably isn't used by many and is thus on par with logging things, the logging crate would require people to use a compatible logging crate to even see the events.

Reply:
I fixed this and update the commit